src/freenet/clients/http/ToadletContextImpl.java

```diff
@@ -440,11 +440,6 @@ static void sendReplyHeaders(
 		mvt.put("x-content-security-policy", contentSecurityPolicy);
 		mvt.put("x-webkit-csp", contentSecurityPolicy);
 		mvt.put("x-frame-options", allowFrames ? "SAMEORIGIN" : "DENY");
-		String HSTS = SSL.getHSTSHeader();
-		if(!HSTS.isEmpty() && !mvt.containsKey("strict-transport-security")) {
-			// SSL enabled, set strict-transport-security so that the user agent upgrade future requests to SSL.
-			mvt.put("strict-transport-security", HSTS);
-		}
```

Comment:
Does this disable HSTS?

Reply:
HSTS headers are already added at line 237 and with proper check of whether fproxy SSL is enabled. But I forgot to delete here.

Reply:
Having both SSL and fproxy SSL configuration is confusing TBH. Especially you can't modify fproxy SSL on the web interface, it looks like SSL configuration is not working.